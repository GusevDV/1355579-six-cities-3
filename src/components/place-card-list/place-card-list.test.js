import React from "react";
import renderer from "react-test-renderer";
import PlaceCardList from "./place-card-list.jsx";
import Offers from '../../test-mocks/offers.js';
import {BrowserRouter} from 'react-router-dom';

it(`Should PlaceCardList component render correctly`, () => {

  const render = renderer.create(
      <BrowserRouter>
        <PlaceCardList
          offers={Offers}
          onMouseEnter={() => {}}
          onMouseLeave={() => {}}
        />
      </BrowserRouter>
  )
  .toJSON();

  expect(render).toMatchSnapshot();
});
