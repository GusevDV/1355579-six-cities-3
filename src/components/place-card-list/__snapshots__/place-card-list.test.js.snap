// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlaceCardList component render correctly 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "50%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="#"
          onClick={[Function]}
        >
          Apartment 1
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            200
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="#"
          onClick={[Function]}
        >
          Apartment 2
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            250
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "95%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="#"
          onClick={[Function]}
        >
          Apartment 3
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            199
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "70%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__name-link"
          href="#"
          onClick={[Function]}
        >
          Apartment 4
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
</div>
`;
